### ISSUE TITLE
no-sequences doesn't work correctly with assignment in arrow function expression

### ISSUE THREAD

**Issue by jramstedt** [eslint/eslint#14572]
Node version: v15.5.0
npm version: v7.3.0
Local ESLint version: v7.25.0     
Global ESLint version: Not found  
Operating System: win32 10.0.19042

@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es2021: true
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module'
  },
  extends: [
    'standard',
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended'
  ],
  plugins: [
    '@typescript-eslint'
  ],
  rules: {
    indent: ['error', 2],
    'no-useless-return': 'off',
    curly: ['error', 'multi-or-nest', 'consistent'],
    'comma-dangle': ['error', 'only-multiline'],
    'no-labels': ['error', { allowLoop: true }],
    'prefer-const': 'error',
    'dot-notation': 'warn',
    '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
    '@typescript-eslint/no-unused-vars': ['warn', { argsIgnorePattern: '^_' }],
    '@typescript-eslint/no-require-imports': 'error',
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let aa = 0
const test = () => ((aa = 123), 10)

console.log(aa, test())
```

```bash
npx eslint test.ts
```

**What did you expect to happen?**
No errors.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
  2:31  error  Unexpected use of comma operator  no-sequences

✖ 1 problem (1 error, 0 warnings)
```

**Steps to reproduce this issue:**

Test the code above.

This code works without errors:
```js
let aa = 0
const test = () => { return ((aa = 123), 10) }
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by nzakas** [comment]
Confirmed, This is a bug. Both examples should have the same result, which is no warnings. The intent is for parentheses to indicate the sequence is intentional ([reference](https://github.com/eslint/eslint/blob/ee3a3ead893d185cc4b1ae9041940cb0968767e1/lib/rules/no-sequences.js#L119)).

**Comment by nzakas** [comment]
@jramstedt do you want to submit a pull request to fix this?

**Comment by jramstedt** [comment]
No.
I don't currently have time to figure this out.

**Comment by snitin315** [comment]
I can take it if there are no objections.

**Comment by mdjermanovic** [comment]
This is intended behavior, per https://github.com/eslint/eslint/issues/6082. 

The rule requires double parentheses around arrow function body:

```js
/* eslint no-sequences: error */

const test1 = () => ((aa = 123), 10); // error

const test2 = () => (((aa = 123), 10)); // ok

```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXNlcXVlbmNlczogZXJyb3IgKi9cblxuY29uc3QgdGVzdDEgPSAoKSA9PiAoKGFhID0gMTIzKSwgMTApOyAvLyBlcnJvclxuXG5jb25zdCB0ZXN0MiA9ICgpID0+ICgoKGFhID0gMTIzKSwgMTApKTsgLy8gb2siLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by nzakas** [comment]
Wow nice catch. This isn’t obvious from the docs for this rule, so I’d like to leave this open to improve the documentation. This special case should be called out explicitly.
